oeps/best-practices/oep-0066-bp-authorization.rst

```diff
@@ -0,0 +1,433 @@
+permitted for all users, the code typically directly checks properties of the
+user: are they a superuser, do they belong to a particular group or have a
+particular role, etc.  This has a few drawbacks:
+
+* This is often a violation of DRY ("Don't Repeat Yourself") which results in
+  the same basic permission check being copied in multiple code locations,
+  making it very difficult to consistently change its implementation (as may
+  happen when a new type of user is introduced, or the need for a special
+  exception becomes clear).
+* Fine-grained permission checks have sometimes been avoided even when
+  appropriate due to the difficulty of copying the permission code around,
+  finding a common place to store it, or updating all the code that used a
+  coarser initial implementation.  This has resulted in some users being
+  technically capable of performing actions which logically should not be
+  permitted.
+* When a decision is made to change who is granted a particular permission,
+  it can be difficult to avoid accidentally changing other permissions with
+  a similar implementation.
+
+In Use System/Protocol Motivation
+----------------------------------
+There are currently multiple systems/protocols that control authorization. 
+Determining which combination of systems/protocols are responsible for granting access 
+is a complex task made more complex by the lack of 
+unified documentation on the systems/protocols. This OEP aims to compile existing 
+knowledge and documentation into a central document that will give an overview of each 
+system/protocol. The aim is not to be the only source of information for each system/protocol, 
+but rather a starting point when learning about authorization within Open edX.
+
+Defined Terms
+*************
+
+Authorization (Authz)
+---------------------
+
+Authorization is the granting of permission of a certain user to perform specific operations in an application. A user can also delegate an application to be authorized to perform operations on their behalf without being logged in or authenticated, which is the basis of OAuth.
+
+.. note::
+  The definition of authorization found here is the same as that found in 
+  :doc:`OEP 42 <oep-0042-bp-authentication>`.
+  Credit for this definition belongs to the authors of OEP-42; Robert Raposa, Nimisha Asthagiri, and Julia Eskew.
+
+Authentication (Authn)
+----------------------
+
+Authentication is the verification of the identity of a user, which typically initiates at a “login” application point. Authentication is required whenever we need to identify a client/caller/user/etc.
+
+.. note::
+
+  Authentication is out of scope of this OEP. 
+  The definition is included here to clarify the difference between it and Authorization.
+  The definition comes from :doc:`OEP 42 <oep-0042-bp-authentication>`.
+  Credit for this definition belongs to the authors of OEP-42; Robert Raposa, Nimisha Asthagiri, and Julia Eskew.
+
+RBAC
+--------------
```

Comment:
```suggestion
----
```

Reply:
In general throughout this doc, the underline length should match the title length. Can you please check on this?